src/tools/tracejob.c

```diff
@@ -403,15 +405,21 @@ parse_log(FILE *fp, char *job, int ind)
 	strcpy(job_buf, job);
 
 	while (fgets(buf, buf_size, fp) != NULL) {
-		while(buf_size == strlen(buf) + 1){
-			buf = (char*)realloc(buf, buf_size * sizeof(char*));
-			tbuf = (char*)realloc(tbuf, buf_size * sizeof(char*));
-			if(buf == NULL || tbuf == NULL || fgets(tbuf, buf_size+1, fp) == NULL){
+		while(buf_size == strlen(buf) + incr){
```

Comment:
I don't understand incr.  The first time through, it's fine because it's 1.  If a buffer is completely full strlen(buf)+1 will be the size of the buffer due to the '\0'.  You increment incr though.  The second time through it'll be 2.  There should be no way strlen(buf)+2 will be the size of a completely full buffer.